test/extra/NotificationTest.cpp

```diff
@@ -9,29 +9,122 @@
 	// Used for setup of all tests. Has access to this instance.
-	TestFixture() = default;
-	~TestFixture() = default;
+	NotificationTest() = default;
+	~NotificationTest() = default;
 
 	// Run once per execution of the test application.
 	// Used only for global setup outside of tests.
 	static void SetUpTestSuite() {}
 	static void TearDownTestSuite() {}
+
+	void buildValidTestTopic();
+	void buildValidDefaultSourceURI();
+	uprotocol::v1::UUri testTopic_;
+	uprotocol::v1::UUri testDefaultSourceURI_;
+	uprotocol::v1::UPayloadFormat format_;
+	std::optional<uprotocol::v1::UPriority> priority_;
+	size_t capture_count_ = 0;
+	uprotocol::v1::UMessage capture_msg_;
+	std::optional<std::chrono::milliseconds> ttl_;
+
+public:
+	void handleCallbackMessage(const uprotocol::v1::UMessage& message);
 };
 
-// TODO replace
-TEST_F(TestFixture, SomeTestName) {}
+void NotificationTest::SetUp() {
+	buildValidDefaultSourceURI();
+	buildValidTestTopic();
+}
+
+void NotificationTest::buildValidDefaultSourceURI() {
+	testDefaultSourceURI_.set_authority_name("10.0.0.1");
+	testDefaultSourceURI_.set_ue_id(0x18000);
+	testDefaultSourceURI_.set_ue_version_major(0x1);
+	testDefaultSourceURI_.set_resource_id(0x0);
+
+	format_ = uprotocol::v1::UPayloadFormat::UPAYLOAD_FORMAT_TEXT;
+	priority_ = uprotocol::v1::UPriority::UPRIORITY_CS1;
+	ttl_ = std::chrono::milliseconds(1000);
+}
+
+void NotificationTest::buildValidTestTopic() {
+	testTopic_.set_authority_name("10.0.0.2");
+	testTopic_.set_ue_id(0x00011101);
+	testTopic_.set_ue_version_major(0xF8);
+	testTopic_.set_resource_id(0x8101);
+}
+
+void NotificationTest::handleCallbackMessage(
+    const uprotocol::v1::UMessage& message) {
+	std::cout << "NotificationTest::handleCallbackMessage\n";
+	capture_msg_ = message;
+	capture_count_++;
+}
+
+TEST_F(NotificationTest, NotificationSuccess) {
+	// Notify Sink
+	auto transportMockNotificationSink =
+	    std::make_shared<uprotocol::test::UTransportMock>(
+	        testDefaultSourceURI_);
+
+	auto callback = [this](const auto& arg) {
+		return this->handleCallbackMessage(arg);
+	};
+
+	std::optional<uprotocol::v1::UUri> source_filter;
+
+	auto result =
+	    NotificationSink::create(transportMockNotificationSink, testTopic_,
+	                             std::move(callback), std::move(source_filter));
+
+	// Notify Source
+	std::string testPayloadStr = "test_payload";
+	auto transportMockNotificationSource_ =
+	    std::make_shared<uprotocol::test::UTransportMock>(
+	        testDefaultSourceURI_);
+
+	auto movableTopic = testTopic_;
+
+	NotificationSource notificationSource(
+	    transportMockNotificationSource_, std::move(movableTopic),
+	    std::move(testDefaultSourceURI_), format_, priority_, ttl_);
+
+	uprotocol::datamodel::builder::Payload testPayload(testPayloadStr, format_);
+	auto status = notificationSource.notify(std::move(testPayload));
+
+	EXPECT_EQ(AsString::serialize(testTopic_),
+	          AsString::serialize(transportMockNotificationSink->sink_filter_));
```

Comment:
```suggestion
	EXPECT_EQ(AsString::serialize(transportMockNotificationSource_->message.sink()),
	          AsString::serialize(transportMockNotificationSink->sink_filter_));
```
Also there should probably be a check for source filters if present

Reply:
Done